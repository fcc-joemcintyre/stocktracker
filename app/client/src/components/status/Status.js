import PropTypes from 'prop-types';

/*
 * Outstanding requests and error status display.
 */
export const Status = ({ requests, errors }) => (
  <div>
    <div>
      Pending requests: {requests.length > 0 ? requests.join (',') : 'none'}
    </div>
    <div>
      Errors: {errors.length > 0 ? errors.join (',') : 'none'}
    </div>
  </div>
);

Status.propTypes = {
  // stock symbols of outstanding requests
  requests: PropTypes.arrayOf (PropTypes.string),
  // errors in format [Stock: symbol Error:# (message)]
  errors: PropTypes.arrayOf (PropTypes.string),
};

Status.defaultProps = {
  requests: [],
  errors: [],
};
