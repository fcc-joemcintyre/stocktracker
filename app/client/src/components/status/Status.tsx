type Props = {
  requests?: string[],
  errors?: string[],
};

/*
 * Outstanding requests and error status display.
 */
export const Status = ({
  requests = [], errors = [],
}: Props) => (
  <div>
    <div>
      Pending requests: {requests.length > 0 ? requests.join (',') : 'none'}
    </div>
    <div>
      Errors: {errors.length > 0 ? errors.join (',') : 'none'}
    </div>
  </div>
);
