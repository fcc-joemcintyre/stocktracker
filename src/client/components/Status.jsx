/**
 * Copyright (c) Joe McIntyre, 2016
 * license: MIT (https://github.com/fcc-joemcintyre/stocktracker/LICENSE.txt)
 */
import React from "react";

/**
 * Outstanding requests and error status display.
 */
export default class Status extends React.Component {
  constructor (props) {
    super (props);
  }

  render() {
    return (
      <div className="statusBox">
        <div>
          Pending requests: {this.props.requests.length > 0
              ? this.props.requests.join (",")
              : "none"}
        </div>
        <div>Errors: {this.props.errors.length > 0
            ? this.props.errors.join (",")
            : "none"}
        </div>
      </div>
    );
  }
}

Status.propTypes = {
  // stock symbols of outstanding requests
  requests: React.PropTypes.arrayOf (React.PropTypes.string),
  // errors in format [Stock: symbol Error:# (message)]
  errors: React.PropTypes.arrayOf (React.PropTypes.string)
}

Status.defaultProps = {
  requests: [],
  errors:[]
}
