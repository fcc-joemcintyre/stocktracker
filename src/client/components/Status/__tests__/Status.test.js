/* eslint-env jest */
import React from 'react';
import { render } from '@testing-library/react';
import { Status } from '../Status';

test ('Status displays default content', () => {
  const { asFragment } = render (
    <Status />
  );
  expect (asFragment).toMatchSnapshot ();
});

test ('Status displays 2 requests', () => {
  const { asFragment } = render (
    <Status requests={['GE', 'V']} />
  );
  expect (asFragment).toMatchSnapshot ();
});

test ('Status displays 2 errors', () => {
  const { asFragment } = render (
    <Status errors={['BCD not found', 'XYZ not found']} />
  );
  expect (asFragment).toMatchSnapshot ();
});

test ('Status displays 2 requests and 2 errors', () => {
  const { asFragment } = render (
    <Status requests={['GE', 'V']} errors={['BCD not found', 'XYZ not found']} />
  );
  expect (asFragment).toMatchSnapshot ();
});
