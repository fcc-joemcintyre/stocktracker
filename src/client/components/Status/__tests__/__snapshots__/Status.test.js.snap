// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status displays 2 errors 1`] = `
<div>
  <div>
    Pending requests: 
    none
  </div>
  <div>
    Errors: 
    BCD not found,XYZ not found
  </div>
</div>
`;

exports[`Status displays 2 requests 1`] = `
<div>
  <div>
    Pending requests: 
    GE,V
  </div>
  <div>
    Errors: 
    none
  </div>
</div>
`;

exports[`Status displays 2 requests and 2 errors 1`] = `
<div>
  <div>
    Pending requests: 
    GE,V
  </div>
  <div>
    Errors: 
    BCD not found,XYZ not found
  </div>
</div>
`;

exports[`Status displays default content 1`] = `
<div>
  <div>
    Pending requests: 
    none
  </div>
  <div>
    Errors: 
    none
  </div>
</div>
`;
